pkg/controller/common/stackmon/config_test.go

```diff
@@ -12,11 +12,122 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/uuid"
 
+	commonv1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/common/v1"
+	esv1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/elasticsearch/v1"
 	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/name"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/stackmon/monitoring"
 	"github.com/elastic/cloud-on-k8s/v2/pkg/utils/k8s"
 )
 
+func Test_newBeatConfig(t *testing.T) {
+	type args struct {
+		initObjects []runtime.Object
+		beatName    string
+		baseConfig  string
+		associated  commonv1.Associated
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    beatConfig
+		wantErr bool
+	}{
+		{
+			name: "Simple output config",
+			args: args{
+				baseConfig: `
+param1: value1
+param2: value2
+`,
+				initObjects: []runtime.Object{
+					&corev1.Secret{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:            "monitored-default-monitoring-beat-es-mon-user",
+							Namespace:       "default",
+							UID:             uuid.NewUUID(),
+							ResourceVersion: "1",
+							Generation:      0,
```

Comment:
`Generation` is indeed pointless (default value). `ResourceVersion` was mostly to avoid the [default `999` from the Fake client](https://github.com/kubernetes-sigs/controller-runtime/blob/2d210d01f344d8c371588564f905185f866f162c/pkg/client/fake/client.go#L195-L201). Regarding `UID` I have no other argument than to have something more "realistic". We can indeed remove them.